### PR TITLE
Counter helper

### LINKED ISSUE #19: Counter helper functions
Keeping track of counters in the events table is hard to write and maintain

- [x] Create helper function that calculates the registered/checkedin/waitlisted count for an event
- [x] Add those values to `events/get`
- [x] Remove counter/incr logic from `registration/create` 
- [x] Add logic for checking if event is full and add user to waitlist instead in `registration/create`

### PR DESCRIPTION
**Pull request by ianmah**
closes #19

### CODE REVIEW

File: serverless.test.yml [hunk 1]
```diff
@@ -3,4 +3,13 @@
...
+      message: "Yeet!"
+      
+      
+- name: event count with 200
+  endpoint: {function: eventCount}
+  request:
+    form:
+      id: bp
+  response:
+    status: 200
```
Comment (jacqueschen1) [inline serverless.test.yml]: Could this test check the contents of the body to be correct?

File: handlers/registration.js [hunk 2]
```diff
@@ -1,77 +1,68 @@
...
-  })
-  .catch(error => {
-    console.error(error);
-    const response = {
-      statusCode: 500,
-      body: error
+    Key: { id: eventID },
+    TableName: 'biztechEvents' + process.env.ENVIRONMENT
+  }
+  // Check if the event is full
```
Comment (jacqueschen1) [inline handlers/registration.js]: I don't think any of the tests covers this case, may be good to add one

Comment (ianmah) [reply handlers/registration.js]: bet

**Comment by ianmah** [comment]
couldn't figure out how to validate the event count body

**Comment by ianmah** [comment]
@jacqueschen1 ready for review when you have a chance

File: handlers/registration.js [hunk 3]
```diff
@@ -27,70 +27,67 @@ module.exports.create = async (event, ctx, callback) => {
...
+  const eventParams = {
+    Key: { id: eventID },
+    TableName: 'biztechEvents' + process.env.ENVIRONMENT
+  }
+  // Check if the event is full
+  await docClient.get(eventParams).promise()
+    .then(async(event) => {
+      const counts = await helpers.getEventCounts(eventID)
+
+      if (counts.registeredCount >= event.Item.capac){
```
Comment (jacqueschen1) [inline handlers/registration.js]: This should only be checked if registrationStatus is registered. If it's cancelled or checkedin, it should just go through.

**Comment by jacqueschen1** [comment]
Right now, registration/create only works if you have "status" and "registrationStatus", shouldn't it be "registrationStatus" on dev?

**Comment by ianmah** [comment]
@jacqueschen1 its because I was deploying the other branch on dev

File: handlers/registration.js (hunk 3 above)
Comment (ianmah) [reply handlers/registration.js]: Good call

**Pull request merged**